### ISSUE TITLE
Can't use other 'paths' aliases in typescript

### ISSUE THREAD

**Issue by luisrudge** [remix-run/remix#2298]
### What version of Remix are you using?

1.2.3

### Steps to Reproduce

repo here: https://github.com/luisrudge/remix-test-paths

Paths config: https://github.com/luisrudge/remix-test-paths/blob/main/tsconfig.json#L15
Trying to use it: https://github.com/luisrudge/remix-test-paths/blob/main/app/root.tsx#L10

### Expected Behavior

I should be able to import an internal library aliased via the 'paths' property in tsconfig.json

### Actual Behavior

```
Error: Cannot find module '@mylib'
Require stack:
- /Users/luisrudge/code/oss/test-paths/api/index.js
- /Users/luisrudge/code/oss/test-paths/node_modules/@remix-run/serve/index.js
- /Users/luisrudge/code/oss/test-paths/node_modules/@remix-run/dev/cli/commands.js
- /Users/luisrudge/code/oss/test-paths/node_modules/@remix-run/dev/cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/luisrudge/code/oss/test-paths/api/route:/Users/luisrudge/code/oss/test-paths/app/root.tsx:10:26)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
```

**Comment by F3n67u** [comment]
remix build your server code via `esbuild` with `serverBareModulesPlugin` as a plugin:

https://github.com/remix-run/remix/blob/f210afccac452e66818a1ae8dfbcac7678e1b6ca/packages/remix-dev/compiler.ts#L419

`serverBareModulesPlugin` mistakenly mark @mylib as a external module because implementation don't take path alias into consideration, so @mylib is not bundled which cause the `Cannot find module '@mylib'` error :

https://github.com/remix-run/remix/blob/main/packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts



**why `~` path alias is work normally?**

because remix process module which path start with `~` specially, so module start with `~` is bundled :

https://github.com/remix-run/remix/blob/f210afccac452e66818a1ae8dfbcac7678e1b6ca/packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts#L25-L28

https://github.com/remix-run/remix/blob/f210afccac452e66818a1ae8dfbcac7678e1b6ca/packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts#L106-L112

**Comment by machour** [comment]
@F3n67u If you have a solution, we'll gladly accept a PR for it

**Comment by F3n67u** [comment]
> @F3n67u If you have a solution, we'll gladly accept a PR for it

@machour  I am very glad to help. I will make a pr tomorrow bec its late now.

**Comment by kiliman** [comment]
The reason the `~` prefix was chosen is that it is uncommon for it to be part of the package name. As you can see in your code snippet above, it checks to see if it's NOT a bare module (in other words, it's not from node_modules), and if not, it will be included in the bundle.

However, the same cannot be said for `@`. That is the npm scope identifier (even Remix uses @remix-run). If you allowed that `@` prefix, then the Remix compiler would end up bundling a lot of packages from node_modules.

I think your PR would need to add some kind of config option to add a list of package prefixes. 
```ts
// remix.config.js
excludeModulePrefixes: ['@myapp']

// serverBareModulesPlugin.ts
function isBareModuleId(id: string): boolean { 
   return ( 
     !id.startsWith("node:") && 
     !id.startsWith(".") && 
     !id.startsWith("~") && 
     !config.excludeModulePrefixes.some(prefix => id.startsWith(prefix)) &&
     !isAbsolute(id) 
   ); 
```

**Comment by machour** [comment]
Fixed by #2412, thank you @F3n67u 🥳
